src/card-product.ts

```diff
@@ -249,4 +253,35 @@ export class CardProductApi {
     return { success: !resp?.payload?.errorCode, body: { ...resp.payload } }
   }
 
+  /*
+   * Function to take some Card Product attributes and update that Card Product
+   * in Marqeta with these values. The return value will be the updated
+   * Card Product.
+   */
+  async update(cardProduct: CardProduct): Promise<{
+    success?: boolean,
+    body?: CardProduct,
+    error?: MarqetaError,
+  }> {
+
+    const resp = await this.client.fire('PUT',
+
+      `cardproducts/${cardProduct?.token}`,
+      undefined,
+      snakecaseKeys(removeEmpty(cleanMetatData(cardProduct))),
```

Comment:
Here's something I was thinking of while walking... if you **_always_** have this code on each `PUT` or `POST` - really any `body`:

```typescript
      snakecaseKeys(removeEmpty(cleanMetatData(cardProduct))),
```

then consider - just consider, putting it in `fire()`... something like:

```typescript
        response = await fetch(url, {
          method: method,
          body: isForm ?
                (body as any) : 
                (body ? JSON.stringify(snakecaseKeys(removeEmpty(cleanMetatData(body)))) : undefined),
          headers,
        })
```

this way you don't have to worry about it... and you don't need to include the functions used only in this line... Just a thought